test/integration/internal_metrics_test.go

```diff
@@ -0,0 +1,178 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+//go:build integration
+
+package integration
+
+import (
+	"net/http"
+	"path"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/mariomac/guara/pkg/test"
+	io_prometheus_client "github.com/prometheus/client_model/go"
+	"github.com/prometheus/common/expfmt"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"go.opentelemetry.io/obi/test/integration/components/docker"
+	"go.opentelemetry.io/obi/test/integration/components/prom"
+)
+
+func TestInstrumentationErrors(t *testing.T) {
```

Comment:
Can we please add a unit test or an integration test that ensures the behaviour or empty version is correct?